backend/typescript/services/interfaces/warningService.ts

```diff
@@ -0,0 +1,36 @@
+export interface WarningDTO {
+  id: number;
+  title: string;
+  description: string;
+  dateIssued?: Date;
+  residentId?: number;
+  assignerId?: number;
```

Comment:
maybe I'm missing something, but why would we want `residentId` and `assignerId` to be optional fields? when creating a warning in Postman, these fields are required, so I don't think the `?` is actually doing anything here, might make sense to remove them.